File: frontend/multilang_examples/hello.pl
greet(Name, Greeting) :-
    format(string(Greeting), "Hello, ~w!", [Name]).
add(A, B, Sum) :-
    Sum is A + B.
:- greet('World', G), writeln(G), add(3, 5, S), format('Sum: ~w~n', [S]).

File: frontend/multilang_examples/hello.rs
fn greet(name: &str) -> String {
    format!("Hello, {}!", name)
}
fn add(a: i32, b: i32) -> i32 {
    a + b
}
fn main() {
    println!("{}", greet("World"));
    println!("Sum: {}", add(3, 5));
}
